packages/ic-http-gateway/tests/protocol_error_tests.rs

```diff
@@ -0,0 +1,94 @@
+use http::{status::StatusCode, Request};
+use ic_agent::{export::Principal, Agent};
+use ic_http_gateway::{HttpGatewayClient, HttpGatewayRequestArgs};
+use reqwest::Client;
+use std::{env, error::Error, process::Command, str::FromStr};
+use testcontainers::{
+    core::{IntoContainerPort, WaitFor},
+    runners::AsyncRunner,
+    GenericImage,
+};
+
+const IMAGE_NAME: &str = "ic-mock-busy-replica";
+const IMAGE_TAG: &str = "latest";
+
+fn build_gateway_image() -> Result<(), Box<dyn Error>> {
+    let cwd = env::var("CARGO_MANIFEST_DIR")?;
+
+    let output = Command::new("docker")
+        .current_dir(format!("{cwd}/test-container"))
+        .arg("build")
+        .arg("--file")
+        .arg("Dockerfile")
+        .arg("--force-rm")
+        .arg("--tag")
+        .arg(format!("{IMAGE_NAME}:{IMAGE_TAG}"))
+        .arg(".")
+        .output()?;
+
+    if !output.status.success() {
+        eprintln!("stderr: {}", String::from_utf8(output.stderr)?);
+        return Err("unable to build no_expose_port:latest".into());
+    }
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_rate_limiting_error() -> Result<(), Box<dyn std::error::Error>> {
+    let res = build_gateway_image();
```

Comment:
Why not just bubble this error up to be consistent with how errors are handled in the rest of the test?
```rust
build_gateway_image()?;
```